src/pkgmt/cli.py

```diff
@@ -174,3 +175,29 @@ def lint():
 
     if returncode:
         raise SystemExit("Error linting")
+
+
+@cli.command()
+@click.option(
+    "-b",
+    "--base-branch",
+    default="main",
+    help="Base branch to compare against",
+)
+@click.option(
+    "-ex",
+    "--exclude-path",
+    default=["doc"],
+    multiple=True,
+    help="Path to exclude from git diff. Can be used multiple times eg: -p p1 -p p2",
```

Comment:
looks like this will evaluate if I changed *anything* except what's listed. but that'd be pretty cumbersome if we want to check whether I *exclusively* changed `doc/` because it'd require me to list all directory. say I have directories a/,..z/ and doc/, if I only want to check if I changed doc/ I need to run:

```
pkgmt ensure-modified --base-branch master -p a -p b -p c ... -p z
```

what we want is to check whether `doc/` is the only one that has been modified by only listing that one:

```
pkgmt ensure-modified --base-branch master --exclusive doc
```

this should fail (exit code 1) if I modified anything outside doc/. and exit with code 0 if I only modified things inside doc/

and this:

```
pkgmt ensure-modified --base-branch master --exclusive doc --exclusive another
```

this would fail if I modified anything outside doc/ or /another

Reply:
True, that is one way to see it.

Another way would be:
Suppose we want to check whether `doc` folder is modified in the codebase, so if I run `pkgmt ensure-modified -b main  -ex doc` and only `doc` folder is modified, excluding it from `git diff` and compare it with `main` branch would result in the output of `0` as **nothing is modified**. But as we know, there is a commit, **so something must be modified(underlying assumption)**, as **excluding doc**, results in no modification in the codebase, therefore **doc must have been modified**

Output: `0` : "**Nothing is modified excluding ...**" is equivalent to "**something is modified in excluded ...**" (Run only doc tests)
`1` : "**Something is modified excluding ...**" is **not** equivalent to "**nothing is modified in excluded ...**" (Run all tests)

@edublancas

Reply:
I'm not following this logic. Let's implement it, test it in dummy and see how it works